### PR TITLE
Dbz 3655

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.redhat.com/browse/DBZ-3655

@jpechane, still there? If so, feedback welcomed :)

### CODE REVIEW

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java [hunk 1]
```diff
@@ -288,10 +293,23 @@ private void searchWalPosition(ChangeEventSourceContext context, final Replicati
...
                 }
             }
+
+            probeConnectionIfNeeded();
         }
         LOGGER.info("WAL resume position '{}' discovered", resumeLsn.get());
     }
 
+    private void probeConnectionIfNeeded() throws SQLException {
+        if (millisSinceLastProbe() > statusUpdateIntervalMs) {
```
Comment (jpechane) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java:303]: How about using `ElapsedTimeStrategy`?

File: debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java [hunk 2]
```diff
@@ -704,6 +704,22 @@ public JdbcConnection prepareQueryWithBlockingConsumer(String preparedQueryStrin
...
+     * Execute a SQL prepared query.
+     *
+     * @param preparedQueryString the prepared query string
+     * @param preparer the function that supplied arguments to the prepared statement; may not be null
+     * @param resultConsumer the consumer of the query results
+     * @return this object for chaining methods together
+     * @throws SQLException if there is an error connecting to the database or executing the statements
+     * @see #execute(Operations)
+     */
+    public JdbcConnection prepareQuery(String preparedQueryString) throws SQLException {
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java:717]: I think JavaDoc is unrelated to the method.

Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java:717]: Mh, yeah, copy and paste. Will fix.

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java (hunk 1 above)
Comment (gunnarmorling) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java:303]: Ha, didn't know of that. Thanks!

**Comment by gunnarmorling** [comment]
Thx for the quick review, @jpechane! Will do the suggested adjustments next week and merge.

**Comment by jpechane** [comment]
@gunnarmorling Applied, thanks!

**Pull request merged**